### ISSUE TITLE
Nixery does not currently support stateful manifest pulls

### ISSUE THREAD

**Issue by mxrlkn** [google/nixery#112]
I'm trying to run arm64 packages on the new M1 Mac as described here: https://github.com/google/nixery/issues/13#issuecomment-546372482

But I get this error when I try to pull an image:

```
docker pull nixery.dev/arm64/shell/git
Using default tag: latest
Error response from daemon: error parsing HTTP 404 response body: unexpected end of JSON input: ""
```

I also tried to pull a non arm64 package with the new docker option `--platform linux/amd64` https://docs.docker.com/docker-for-mac/apple-m1.

but that doesn't work either.

**Comment by enblanco** [comment]
Same problem here. Even with the quick demo on the homepage:

`$ docker pull nixery.dev/shell/git/htop
Using default tag: latest
Error response from daemon: error parsing HTTP 404 response body: unexpected end of JSON input: ""`

**Comment by blakehawkins** [comment]
I have the same problem on WSL2

```
$ uname -r
4.19.104-microsoft-standard

$ docker pull nixery.dev/shell/git/htop
Using default tag: latest
Error response from daemon: error parsing HTTP 404 response body: unexpected end of JSON input: ""
```

**Comment by tazjin** [comment]
Changed the title slightly to accommodate the actual issue here:

Docker has switched by default from just GETing the manifest to issuing a HEAD request and then fetching the resulting manifest by its content hash again. Nixery does not currently support this, and it needs to be implemented.

Example request: `GET https://nixery.dev/v2/shell/git/htop/manifests/sha256:8dd204e4436b23aeedf76bd0b6144b18b8913316c2fa6d9e060b3eeafea6a602`

**Comment by tazjin** [comment]
I expected this to be fixed by the patches in #105, but it's not the case. Going to investigate.

**Comment by tazjin** [comment]
This should be resolved now, please reopen the issue if you still see it. The problem was that some requests to nixery.dev where hitting an outdated version of the backend, which has now been turned down.
